bitcoin/src/consensus/encode.rs

```diff
@@ -136,6 +136,30 @@ pub fn deserialize_partial<T: Decodable>(data: &[u8]) -> Result<(T, usize), Erro
     Ok((rv, consumed))
 }
 
+/// A trait for integer types smaller than 64-bits.
+pub trait IntoU64 {
+    /// Convert into a u64.
+    fn into_u64(self) -> u64;
+}
+
+impl IntoU64 for u8 {
+    fn into_u64(self) -> u64 { self as u64 }
+}
+impl IntoU64 for u16 {
+    fn into_u64(self) -> u64 { self as u64 }
+}
+impl IntoU64 for u32 {
+    fn into_u64(self) -> u64 { self as u64 }
+}
+impl IntoU64 for u64 {
+    fn into_u64(self) -> u64 { self }
+}
+//TODO(stevenroose) do we have to do some compile-time checks
+// that we're not using a 128-bit architecture?
```

Comment:
I added a `NB` instead, do you think that's enough at this point? Or should I add a unit tests in this module specifically to check that usize doesn't exceed 64 bits?

Reply:
It should be more along the lines of:

```rust
#[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
compile_error!("Only 32 and 64 bit architectures are supported");
```

Which I think we have somewhere but it's far away. Ideally we would have a module (or extension trait) with conversion functions that contains that check and then we would call into those everywhere and get rid of all `as` or `try_into().unwrap()` conversions in the whole codebase.

But for now `NB` is fine since the code is probably littered with these things anyway.